src/openvic-simulation/economy/production/ArtisanalProducerFactoryPattern.cpp

```diff
@@ -0,0 +1,45 @@
+#include "ArtisanalProducerFactoryPattern.hpp"
+
+#include "openvic-simulation/economy/GoodInstance.hpp"
+#include "economy/production/ArtisanalProducer.hpp"
+
+using namespace OpenVic;
+
+ArtisanalProducerFactoryPattern::ArtisanalProducerFactoryPattern(
+	MarketInstance& new_market_instance,
+	ModifierEffectCache const& new_modifier_effect_cache,
+	ProductionTypeManager const& new_production_type_manager
+) : market_instance { new_market_instance },
+	modifier_effect_cache { new_modifier_effect_cache },
+	production_type_manager { new_production_type_manager }
+	{}
+
+std::unique_ptr<ArtisanalProducer> ArtisanalProducerFactoryPattern::CreateNewArtisanalProducer(Pop& pop) const {
+	ProductionType const* random_artisanal_production_type = nullptr;
+
+	for (ProductionType const& production_type : production_type_manager.get_production_types()) {
+		if (production_type.get_template_type() == ProductionType::template_type_t::ARTISAN) {
+			GoodInstance const& good_instance = market_instance.get_good_instance_manager().get_good_instance_from_definition(
+				production_type.get_output_good()
+			);
+			if (!good_instance.get_is_available()) {
```

Comment:
Are artisanal production types unlocked globally or per country?
If per country, we need to add that here.